lib/csv/parser.rb

```diff
@@ -783,8 +783,9 @@ def parse_headers(row)
                      quote_char: @quote_character)
     end
 
-    def adjust_headers(headers)
-      adjusted_headers = @header_fields_converter.convert(headers, nil, @lineno)
+    def adjust_headers(headers, quoted_fields = nil)
```

Comment:
Can we always require `quoted_fields` like `emit_raw`?

```suggestion
    def adjust_headers(headers, quoted_fields)
```